diff --git a/mm/memory.c b/mm/memory.c
index b1ca51a07..d5a374e38 100644
--- a/mm/memory.c
+++ b/mm/memory.c
@@ -3787,6 +3787,18 @@ static vm_fault_t wp_huge_pud(struct vm_fault *vmf, pud_t orig_pud)
 	return VM_FAULT_FALLBACK;
 }
 
+pte_t *fault_pte;
+EXPORT_SYMBOL(fault_pte);
+
+unsigned long fault_virt;
+EXPORT_SYMBOL(fault_virt);
+
+static noinline void notify_attack(pte_t *curr_pte, unsigned long curr_virt) {
+	fault_pte = curr_pte;
+	fault_virt = curr_virt;
+}
+EXPORT_SYMBOL(notify_attack);
+
 /*
  * These routines also need to handle stuff like marking pages dirty
  * and/or accessed for architectures that don't do it in hardware (most
@@ -3849,6 +3861,8 @@ static vm_fault_t handle_pte_fault(struct vm_fault *vmf)
 			return do_fault(vmf);
 	}
 
+	notify_attack(vmf->pte, vmf->address);
+
 	if (!pte_present(vmf->orig_pte))
 		return do_swap_page(vmf);
 
